python/ml4ir/model/ranking_model.py

```diff
@@ -351,32 +424,20 @@ def _predict_score(features, label):
 
         return _predict_score
 
-    def evaluate(self, test_dataset, models_dir, logs_dir, logging_frequency=25):
-        """
-        Predict the labels/ranks and compute metrics for the test set
-
-        Args:
-            test_dataset: an instance of tf.data.dataset
+    def _convert_predictions_to_df(self, predictions_dict, features):
+        # If feature is a string, convert back from bytes to string
+        for feature_info in features:
+            feature_node_name = feature_info.get("node_name", feature_info["name"])
+            if feature_info["feature_layer_info"]["type"] == FeatureTypeKey.STRING:
+                str_feature = tf.strings.unicode_encode(
+                    tf.cast(predictions_dict[feature_node_name], tf.int32),
+                    output_encoding="UTF-8",
+                )
+                predictions_dict[feature_node_name] = tf.strings.regex_replace(
+                    str_feature, "\x00", ""
```

Comment:
what is `\x00` in python?

Reply:
Actually I am not sure what this is. This is the default padding character on tensorflow.

Reply:
I should leave a TODO for the future to fix this. Nothing critical though.